src/Context.php

```diff
@@ -58,198 +58,64 @@ public function withData(array $data, bool $keepExisting = true, bool $recursive
 
         if ($keepExisting) {
             if ($recursive) {
-                /* @var array<(int|string), mixed> */
                 $data = $this->arrayMergeRecursiveDistinct($this->data, $data);
             } else {
-                /* @var array<(int|string), mixed> */
                 $data = array_merge($this->data, $data);
             }
         }
 
-        return new self(
-            $data,
-            $this->environment,
-            $this->currentDirectory,
-            $this->tty,
-            $this->pty,
-            $this->timeout,
-            $this->quiet,
-            $this->allowFailure,
-            $this->notify,
-            $this->verbosityLevel,
-            $this->name,
-        );
+        return $this->with(data: $data);
     }
 
     /** @param array<string, string|\Stringable|int> $environment */
     public function withEnvironment(array $environment, bool $keepExisting = true): self
     {
-        return new self(
-            $this->data,
-            $keepExisting ? [...$this->environment, ...$environment] : $environment,
-            $this->currentDirectory,
-            $this->tty,
-            $this->pty,
-            $this->timeout,
-            $this->quiet,
-            $this->allowFailure,
-            $this->notify,
-            $this->verbosityLevel,
-            $this->name,
-        );
+        return $this->with(environment: $keepExisting ? [...$this->environment, ...$environment] : $environment);
     }
 
     public function withPath(string $path): self
     {
-        return new self(
-            $this->data,
-            $this->environment,
-            str_starts_with($path, '/') ? $path : PathHelper::realpath($this->currentDirectory . '/' . $path),
-            $this->tty,
-            $this->pty,
-            $this->timeout,
-            $this->quiet,
-            $this->allowFailure,
-            $this->notify,
-            $this->verbosityLevel,
-            $this->name,
-        );
+        return $this->with(currentDirectory: str_starts_with($path, '/') ? $path : PathHelper::realpath($this->currentDirectory . '/' . $path));
```

Comment:
I notice something I don't like.
`withPath` but the property is `currentDirectory`.
What about deprecating `withPath` to `withCurrentDirectory`? 
(if okay, I'll open a new PR)

Reply:
Ok for renaming it, it bugged me a few times :+1:

Reply:
The current name of the `run` method for defining the path is also `path`. Shouldn't we also make sure they're identical?